### PR TITLE
Add support for React routes

### PR DESCRIPTION
**Pull request by marcja**
Add support for React routes by adding a new CallMeBackControlller and explicitly listing the React routes that require redirection.

### CODE REVIEW

File: callmeback/src/main/java/org/google/callmeback/CallMeBackController.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package org.google.callmeback;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+ 
+@Controller
+public class CallMeBackController {
```
Comment (kebrahim) [inline callmeback/src/main/java/org/google/callmeback/CallMeBackController.java:7]: does https://github.com/google-org/new-project-ny/blob/master/callmeback/frontend/src/Routes.jsx need to be updated/removed?

**Comment by marcja** [comment]
So the formatting issue was unrelated to this PR and due to https://github.com/google-org/new-project-ny/issues/75.

File: callmeback/src/main/java/org/google/callmeback/CallMeBackController.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ 
+    @RequestMapping(value = {
```
Comment (kebrahim) [inline callmeback/src/main/java/org/google/callmeback/CallMeBackController.java:9]: do we need to explicitly call out "/", like we do in Routes.jsx? https://github.com/google-org/new-project-ny/blob/master/callmeback/frontend/src/Routes.jsx#L19

File: callmeback/src/main/java/org/google/callmeback/CallMeBackController.java (hunk 1 above)
Comment (mscharfstein) [reply callmeback/src/main/java/org/google/callmeback/CallMeBackController.java:7]: hm ya, +1. Don't have a good enough understanding of how this works, but should the FE routing stay how it is?

File: callmeback/src/main/java/org/google/callmeback/CallMeBackController.java (hunk 2 above)
Comment (mscharfstein) [reply callmeback/src/main/java/org/google/callmeback/CallMeBackController.java:9]: looks like everything gets mapped to "/" so we probably don't need to include "/" in the mapping?

File: callmeback/src/main/java/org/google/callmeback/CallMeBackController.java (hunk 1 above)
Comment (marcja) [reply callmeback/src/main/java/org/google/callmeback/CallMeBackController.java:7]: Yes, the client-side routing stays the same. These two things are accomplishing different things.

The client-side routing that React is doing is purely client-side. When you navigate from one route to another (such as /home to /reservations) within React, the server isn't involved at all. React is interpreting the routes internally. 

The problem is when you either browser refresh on a route (such as /reservations) or navigate directly to that route (such as from a link outside of React). In this case, the server (not React) sees a url like /reservations and doesn't normally know what to do with it, so it returns 404. This PR tells the server, "Hey, when you see a path like /reservations, just redirect to "/". This means that index.html gets served, React gets a chance to boot, and then it takes over handling of the /reservations route.

File: callmeback/src/main/java/org/google/callmeback/CallMeBackController.java (hunk 2 above)
Comment (marcja) [reply callmeback/src/main/java/org/google/callmeback/CallMeBackController.java:9]: No. a) "/" is fine and has the correct behavior already (that is, server up index.html). b) If I added that, server would get into a circular loop trying to resolve "/" to "/" _ad infinitum_.

File: callmeback/src/main/java/org/google/callmeback/CallMeBackController.java (hunk 1 above)
Comment (kebrahim) [reply callmeback/src/main/java/org/google/callmeback/CallMeBackController.java:7]: sounds good, mind just adding a comment here w/ 1-2 sentences explaining that & also mentioning that this file should be kept consistent w/ Routes.jsx?

**Pull request merged**